packages/location_permissions/android/src/main/java/com/baseflow/location_permissions/LocationPermissionsPlugin.java

```diff
@@ -169,12 +168,20 @@ public void onMethodCall(MethodCall call, Result result) {
 
   @Override
   public void onListen(Object arguments, EventSink events) {
-    mEventSink = events;
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+      getActiveContext().registerReceiver(mReceiver, mIntentFilter);
```

Comment:
I think we should save the context in a class member here to make sure we unsubscribe from the same context when we cancel listening to the stream. Depending on the state of the App it is possible that the `getActiveContext` returns a different value (e.g. depending if the App is active of in the background). Something like:

```
mContext = getActiveContext();
mContext.registerReceiver(mReceiver, mIntentFilter);
```

Reply:
Wouldn't this introduce a potential memory leak? Sorry if this is a misinformed question, but I was under the impression keeping references to `Context` instances was bad practice.

Reply:
You are right indeed, thank you for letting me know.

According to this [article](https://android-developers.googleblog.com/2009/01/avoiding-memory-leaks.html?m=1) we should use the ApplicationContext which is a singleton so we will always have the same instance.

We can access the ApplicationContext through the `mRegistrar.context()` method (see also: https://docs.flutter.io/javadoc/io/flutter/plugin/common/PluginRegistry.Registrar.html).

So I think it would be good to replace `getActiveContext()` with `mRegistrar.context()`

Reply:
You are right indeed, thank you for letting me know.

According to this [article](https://android-developers.googleblog.com/2009/01/avoiding-memory-leaks.html?m=1) we should use the ApplicationContext which is a singleton so we will always have the same instance.

We can access the ApplicationContext through the `mRegistrar.context()` method (see also: https://docs.flutter.io/javadoc/io/flutter/plugin/common/PluginRegistry.Registrar.html).

So I think it would be good to replace `getActiveContext()` with `mRegistrar.context()`

Reply:
Updated the implementation accordingly!